spec/models/claim/matching_attribute_finder_spec.rb

```diff
@@ -2,24 +2,27 @@
 
 RSpec.describe Claim::MatchingAttributeFinder do
   describe "#matching_claims" do
-    let(:source_claim) {
-      create(
-        :claim,
+    let(:claim_attributes) {
+      {
         teacher_reference_number: "0902344",
         national_insurance_number: "QQ891011C",
         email_address: "genghis.khan@mongol-empire.com",
         bank_account_number: "34682151",
         bank_sort_code: "972654",
-        building_society_roll_number: "123456789/ABCD"
-      )
+        building_society_roll_number: "123456789/ABCD",
+        policy: StudentLoans
+      }
     }
 
+    let(:source_claim) { create(:claim, claim_attributes) }
+
+    let!(:matching_claim_with_different_policy) { create(:claim, :submitted, claim_attributes.merge(policy: MathsAndPhysics)) }
     let!(:claim_with_no_matching_attributes) { create(:claim, :submitted) }
     let!(:unsubmitted_claim_with_matching_teacher_reference_number) { create(:claim, :submittable, teacher_reference_number: source_claim.teacher_reference_number) }
 
     subject(:matching_claims) { Claim::MatchingAttributeFinder.new(source_claim).matching_claims }
 
-    it "does not include the source claim, or claims that do not match, or claims that are not submitted" do
+    it "does not include the source claim, or claims that do not match, or claims that are not submitted, or claims that match, but have a different policy" do
```

Comment:
This spec has got a bit smelly – it's trying to communicate too many things in a single spec/line. How about we break out three focused specs for each? The three records that are being created in the `let!` calls could be created inline on each of the specs instead of being created for every single spec in this file.

Reply:
👍 Sounds like a plan